src/coreclr/jit/lclmasks.cpp

```diff
@@ -0,0 +1,480 @@
+        LclMasksWeight weight;
+        bool           found = weightsTable->Lookup(lclOp->GetLclNum(), &weight);
+        assert(found);
+
+        // Quit if the cost of changing is higher.
+        if (weight.currentCost <= weight.switchCost)
+        {
+            JITDUMP("Local %s V%02d at [%06u] will not be converted. ", isLocalStore ? "store" : "var",
+                    lclOp->GetLclNum(), Compiler::dspTreeID(lclOp));
+            weight.DumpTotalWeight();
+            return fgWalkResult::WALK_CONTINUE;
+        }
+
+        JITDUMP("Local %s V%02d at [%06u] will be converted. ", isLocalStore ? "store" : "var", lclOp->GetLclNum(),
+                Compiler::dspTreeID(lclOp));
+        weight.DumpTotalWeight();
+
+        // Fix up the type of the lcl and the lclvar.
+        assert(lclOp->gtType != TYP_MASK);
+        var_types lclOrigType = lclOp->gtType;
+        lclOp->gtType         = TYP_MASK;
+        LclVarDsc* varDsc     = m_compiler->lvaGetDesc(lclOp->GetLclNum());
+        varDsc->lvType        = TYP_MASK;
+
+        // Add or remove a conversion
+
+        if (isLocalStore && removeConversion)
+        {
+            // Convert
+            //      use:STORE_LCL_VAR(ConvertMaskToVector(x))
+            // to
+            //      use:STORE_LCL_VAR(x)
+
+            GenTreeHWIntrinsic* convertOp = lclOp->Data()->AsHWIntrinsic();
+
+            lclOp->gtOp1 = convertOp->Op(1);
+            convertOp->gtBashToNOP();
+            m_compiler->fgSequenceLocals(stmt);
+        }
+
+        else if (isLocalStore && addConversion)
+        {
+            // Convert
+            //      use:STORE_LCL_VAR(x)
+            // to
+            //      use:STORE_LCL_VAR(ConvertVectorToMask(x))
+
+            // There is not enough information in the lcl to get simd types. Instead reuse the cached
+            // simd types from the removed convert nodes.
+            assert(weight.simdBaseJitType != CORINFO_TYPE_UNDEF);
+            GenTree* convertOp = m_compiler->gtNewSimdCvtVectorToMaskNode(TYP_MASK, lclOp->Data(),
+                                                                          weight.simdBaseJitType, weight.simdSize);
+            lclOp->Data()      = convertOp;
+        }
+
+        else if (isLocalUse && removeConversion)
+        {
+            // Convert
+            //      user(use:ConvertVectorToMask(LCL_VAR(x)))
+            // to
+            //      user(use:LCL_VAR(x))
+
+            GenTree* const convertOp = *use;
+
+            // Find the location of convertOp in the user
+            int opNum = 1;
+            for (; opNum <= user->AsHWIntrinsic()->GetOperandCount(); opNum++)
+            {
+                if (user->AsHWIntrinsic()->Op(opNum) == convertOp)
+                {
+                    break;
+                }
+            }
+            assert(opNum <= user->AsHWIntrinsic()->GetOperandCount());
+
+            // Remove the convert convertOp
+            convertOp->gtBashToNOP();
```

Comment:
```suggestion
```

Reply:
I'm not sure what you're suggesting to change here - every line was identical when I copy/pasted it.

Reply:
All this code looks unnecessary -- I think it can be deleted

Reply:
> I think it can be deleted

Yes! - it's left over code from when the logic was more complicate